src/reconcile/rec_visibility.c

```diff
@@ -1247,12 +1248,19 @@ __rec_fill_tw_from_upd_select(WT_SESSION_IMPL *session, WT_PAGE *page, WT_CELL_U
         /*
          * Find the update this tombstone applies to.
          *
-         * We need to find the full update if the prepared tombstone is rolled back. We resolve
-         * prepare updates recursively from the oldest to the newest. If we write a prepared
-         * tombstone, we may see a prepared update that is rolled back from the same transaction.
-         * Ensure we also write it to the disk in this case.
+         * To handle scenarios where a prepared tombstone is rolled back, we need to retrieve the
+         * full update associated with it. We resolve prepared updates recursively, processing them
+         * from the oldest to the newest. If a prepared tombstone is written, there's a possibility
+         * that a prepared update from the same transaction was rolled back. In such cases, ensure
+         * that the rolled-back update is also written to disk to maintain data consistency.
+         *
+         * Additionally, if a tombstone with a zero timestamp is selected, it is critical to
+         * identify the update it deletes. Failure to do so may result in missed deletions of
+         * corresponding updates in the history store, leading to potential inconsistencies.
```

Comment:
It looks like this comment addresses this part of the check: `upd->upd_start_ts == WT_TS_NONE`. Should you address `F_ISSET(r, WT_REC_HS)`?

I think the addition of `tombstone_globally_visible` is very good for clarity. We should have one for `(F_ISSET(r, WT_REC_HS) && upd->upd_start_ts == WT_TS_NONE)` as well. And you can declare those variables here by the way.

Reply:
Thank you, I have suggested a comment edit to reflect this.